doc/TabletRouting.md

```diff
@@ -78,28 +78,25 @@ This module also provides helper methods to find a tablet to route traffic
 to. Since it knows the health status of all tablets for a given keyspace / shard
 / tablet type, and their replication lag, it can provide a good list of tablets.
 
-## Vtgate Gateway interface
+## Vtgate TabletGateway
 
-An important interface inside vtgate is the Gateway interface. It can send
+An important interface inside vtgate is the TabletGateway. It can send
 queries to a tablet by keyspace, shard, and tablet type.
 
 As mentioned previously, the higher levels inside vtgate can resolve queries to
-a keyspace, shard and tablet type. The queries are then passed to the Gateway
-implementation inside vtgate, to route them to the right tablet.
+a keyspace, shard and tablet type. The queries are then passed to the TabletGateway inside vtgate,
+to route them to the right tablet.
 
-There are two implementations of the Gateway interface:
-
-* discoveryGateway: deprecated
-* tabletGateway: Combines a set of TopologyWatchers (described in the
-  discovery section, one per cell) as a source of tablets, a HealthCheck module
-  to watch their health, and a tabletHealthCheck per tablet to collect all the health
-  information. Based on this data, it can find the best tablet to use.
+The TabletGateway Combines a set of TopologyWatchers (described in the
+discovery section, one per cell) as a source of tablets, a HealthCheck module
+to watch their health, and a tabletHealthCheck per tablet to collect all the health
+information. Based on this data, it can find the best tablet to use.
   
 # Extensions, work in progress
 
 ## Regions, cross-cell targeting
 
-At first, the discoveryGateway was routing queries the following way:
+At first, the gateway was routing queries the following way:
```

Comment:
We need to rewrite this section. We don't have regions or l2vtgate any more.
Instead we have cell-aliases that group cells together.

Reply:
The entire paragraph on l2vtgate can be deleted.
The section on Region should be either deleted, or rewritten to talk about cell-aliases.
I'm fine with either approach.
We will need to change the heading that says "Regions, ..."

Reply:
Ref: https://github.com/vitessio/vitess/pull/4767

Reply:
I removed both sections via 0785e4a